sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -870,14 +870,14 @@ impl SocRegistersImpl {
 
         for _ in (0..n).step_by(core::mem::size_of::<u32>()) {
             let buf = self.mailbox.read_dataout().unwrap();
-            file.write(&buf.to_le_bytes()).unwrap();
+            file.write_all(&buf.to_le_bytes()).unwrap();
```

Comment:
Not for this PR, but this function is `unwrap()` heavy.

Reply:
I'm not convinced this function should even exist.